openlcsd/flow/tasks.py

```diff
@@ -753,12 +770,68 @@ def save_container_components(context, engine):
     engine.logger.info('Finished saving container components.')
 
 
-def get_component_source_path(context, engine):
-    raise NotImplementedError
+def get_components_product_from_source_container(context, engine):
+    engine.logger.info('Start to get components data from source container...')
+    config = context.get('config')
+    sc_nvr = context.get('package_nvr')
+    # Get components from the source conainer ifself.
+    srpm_dir = context.get('srpm_dir')
+    misc_dir = context.get('misc_dir')
+    sc_handler = SourceContainerHandler(config)
+    components = sc_handler.get_container_components(srpm_dir, misc_dir,
+                                                     sc_nvr)
+    engine.logger.info('Finished getting components from source container')
+    context['components'] = components
+
+
+def import_types_components(context, engine, nvr_list, src_dir, comp_type):
+    """
+    In a source container, it has different type of component. The different
+    components have different src_dir. This founction will be define the post
+    data for the differnt type component.
+    """
+    cli = context.get('client')
+    scan_options = {
+        'crypto_scan': context.get('crypto_scan'),
```

Comment:
since these are boolean values, you may want to provide default other than `None` in case it's not in `context`.

Reply:
Remove the options that could be gotten from the default values.